libs/app-theme/ui/src/lib/file-preview/file-preview-error.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { ButtonType, IconSize, IconType } from "app-theme/models"
+import { Button } from "../button/button"
+import { Typography } from "../typography/typography"
+import { ComponentProps, FunctionComponent } from "react"
+import styled from "styled-components"
+import { Icon } from "../icon/icon"
+import { motion } from "motion/react"
+
+export enum FilePreviewErrorCode {
+  NotFound = 404,
+  Unknown = 500,
+  UnsupportedFileType = 501,
+}
+
+interface Props extends ComponentProps<typeof ErrorWrapper> {
+  error: FilePreviewErrorCode
+  fileType?: string
+  onClose?: VoidFunction
+  onRetry?: VoidFunction
+}
+
+export const FilePreviewError: FunctionComponent<Props> = ({
```

Comment:
Maybe we could move the error branching to the parent and pass ready text/handlers here, so this stays a dumb/presentational component?

Reply:
FilePreview doesn't need to know about the internals of FilePreviewError, so I'd leave it here. Also the error messages are kinda generic so no need to make them customizable.